zipkin-datadog-reporter-core/src/main/java/io/smartup/zipkin/DatadogReporter.java

```diff
@@ -23,30 +23,23 @@
 public class DatadogReporter implements Reporter<Span>, Flushable, Closeable {
     public static final long TIMEOUT_DELAY = TimeUnit.SECONDS.toNanos(30);
     public static final long COMPLETION_DELAY = TimeUnit.SECONDS.toNanos(1);
-    public static final long FLUSH_DELAY = TimeUnit.SECONDS.toMillis(1);
 
     private final Queue<List<DDMappingSpan>> reportingTraces;
+    private final long messageTimeout;
     private final Map<String, PendingTrace> pendingTraces = new ConcurrentHashMap<>();
 
     private final AtomicBoolean running = new AtomicBoolean(false);
     private final DDApi ddApi;
     private volatile Thread flushingThread;
 
-    /**
-     * Report traces to the Datadog Agent at the default location (localhost:8126).
-     */
-    public DatadogReporter() {
-        this(DDApi.DEFAULT_HOSTNAME, DDApi.DEFAULT_PORT, new LinkedBlockingQueue<>());
-    }
-
     /**
      * Report traces to the configured Datadog Agent.
      *
      * @param host (See DDApi.DEFAULT_HOSTNAME)
      * @param port (See DDApi.DEFAULT_PORT)
      */
-    public DatadogReporter(String host, int port) {
-        this(host, port, new LinkedBlockingQueue<>());
+    public DatadogReporter(String host, int port, int messageTimeout) {
```

Comment:
Maybe it would be better to pass the entire `DatadogProperties` class to the constructor. Also have a default constructor that creates a `DatadogProperties` object with the default values.

Reply:
@ambrusadrianz That would be nice, but unfortunately would require establishing a dependency from this module to "zipkin-datadog-reporter-autoconfigure". I think that's the reason why the default Zipkin reporter for Spring Sleuth also implements the constructor this way.